Source/ToolCore/Build/AndroidProjectGenerator.cpp

```diff
@@ -281,6 +313,55 @@ bool AndroidProjectGenerator::GenerateAndroidManifest()
 
 }
 
+bool AndroidProjectGenerator::CopyUserIcons()
+{
+    FileSystem* fileSystem = GetSubsystem<FileSystem>();
+    ToolSystem* toolSystem = GetSubsystem<ToolSystem>();
+    Project* project = toolSystem->GetProject();
+    AndroidBuildSettings* settings = project->GetBuildSettings()->GetAndroidBuildSettings();
+
+    String userIconPath = settings->GetIconPath();
+    if (!fileSystem->DirExists(userIconPath))               // dont do anything if there is no path defined.
+        return true;
+            
+    String userIconDir = userIconPath + "/drawable";        // 1st target dir 
+    String userIconFile = userIconDir + "/logo_large.png";  // 1st target file
+    String destDir = buildPath_ + "/res/drawable";          // where it should be in the build
+    if ( fileSystem->FileExists (userIconFile) )            // is there a file there?
+    {
+        Poco::File icon1 ( userIconFile.CString() );        // declare the file
+        icon1.copyTo( destDir.CString() );                  // copy the file into the build
```

Comment:
Oh my, BuildBase.cpp is quite the treasure trove.  This may have to be a refactor job. The BuildBase file operator methods are in the protected area of the class, so they will have to get moveed into the public area to be usable passed into the Android project generator. Not a lot of risk to do this. I'll audit the other builders also, and update them, I see some are not using the BuildBase file ops, if its not done now, it'll never be done. And it will provide a good examples for the BuildLinux module.
The call to fileSystem->RemoveDir(buildPath_, true); is recursive crash code on linux. This is called when you attempt to create a project where one already exists. Windows does not use this code. The Poco File::remove(true) does work. If RemoveDir wont work, I can ifdef it for linux, and test on OSX.